### ISSUE TITLE
Unable to use multiremote feature Error: Invalid or unsupported WebDriver capabilities found ("protocol", "hostname", "port", "path"). Ensure to only use valid W3C WebDriver capabilities

### ISSUE THREAD

**Issue by Amrkamel1** [webdriverio/webdriverio#6445]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 7.0.3
 - **Mode:** [Standalone mode or WDIO Testrunner] WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** [e.g. sync/async] sync
 - **Node.js version:** 14.15.4
 - **NPM version:** 6.14.10
 - **Browser name and version:** chrome 88.0.4324.150
 - **Platform name and version:** mac big sur
 - **Additional wdio packages used (if applicable):**  all package below are latest
   @wdio/cli
   wdio-chromedriver-service
  @wdio/appium-service
  wdio/mocha-framework
  wdio/local-runner
  wdio/selenium-standalone-service
  wdio/sync
 
**Config of WebdriverIO**
    
```
          capabilities: caps.CrossCaps,
      baseUrl: 'example.com',
      services: [
        ['selenium-standalone', {
          installArgs: {
            version : "3.141.59",
            baseURL : "https://selenium-release.storage.googleapis.com",
            drivers : {
                chrome : {
                    version : "87.0.4280.20",
                    arch    : process.arch,
                    baseURL : "https://chromedriver.storage.googleapis.com"
                }
            }
          },
            args: {
              seleniumArgs: ['-host','localhost','-port', '4444'],
              drivers
            },
        }],  
        // ['browserstack'],
        ['appium',{
          command : 'appium',
          args: {
              debugLogSpacing: true,
              sessionOverride: true,
              port: 4728,
              // allowInsecure:'chromedriver_autodownload'
          },
        }]
      ],
```

```
    caps.ChromeDesktopCaps = 
{
    protocol: 'http',
    hostname: 'localhost',
    port: 4444,
    path: '/wd/hub',
    capabilities: {
        browserName: 'chrome',
        acceptInsecureCerts: true,
        'goog:chromeOptions': {
          args: [
          'user-data-dir=/home/amrka/.config/google-chrome/Default',
          '--disable-web-security',
            'start-maximized',
          ],
        }
    }
  }

caps.AndroidXCaps =
{
    port: 4728,
    path: '/',
    capabilities: {
      platformName: 'Android',
      'appium:deviceName': 'pixel2',
      'appium:avd': 'pixel2',
      'appium:app': 'src/test/resources/test.apk',
      'appium:platformVersion': '10',
      'appium:avdLaunchTimeout': 120000,
      'appium:avdReadyTimeout': 120000,
      'appium:uiautomator2ServerInstallTimeout' :120000,
      'appium:appWaitForLaunch': true,
      'appium:ensureWebviewsHavePages': true,
      'appium:disableWindowAnimation': true,
      'appium:autoGrantPermissions': true,
      'appium:clearSystemFiles': true,
      'appium:nativeWebScreenshot': true,
      'chromedriverExecutable': 'src/test/resources/chromedriver'
    }
}
```

**Describe the bug**

**To Reproduce**
- Upgrade wdio to latest version (v7)
- set multiremote feature as shown in the config above (it was working on previous version)
- execute the test 

**Expected behavior**
should be able to launch the two services instance normally

**Log**

```
    Execution of 1 spec files started at 2021-02-15T17:29:07.711Z

2021-02-15T17:29:07.713Z DEBUG @wdio/utils:initialiseServices: initialise service "selenium-standalone" as NPM package
2021-02-15T17:29:07.713Z DEBUG @wdio/utils:initialiseServices: initialise service "appium" as NPM package
2021-02-15T17:29:07.714Z INFO @wdio/cli:launcher: Run onPrepare hook
2021-02-15T17:29:07.716Z DEBUG @wdio/appium-service: Will spawn Appium process: appium --base-path / --debug-log-spacing --session-override --port 4728
2021-02-15T17:29:08.255Z DEBUG @wdio/appium-service: Appium started with ID: 2206
2021-02-15T17:29:09.142Z DEBUG @wdio/cli:utils: Finished to run "onPrepare" hook in 1428ms
2021-02-15T17:29:09.145Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2021-02-15T17:29:09.145Z DEBUG @wdio/cli:utils: Finished to run "onWorkerStart" hook in 0ms
2021-02-15T17:29:09.146Z INFO @wdio/local-runner: Start worker 0-0 with arg: src/test/config/xplatform.conf.js
[0-0] 2021-02-15T17:29:09.407Z INFO @wdio/local-runner: Run worker command: run
[0-0] 2021-02-15T17:29:10.081Z DEBUG @wdio/config:utils: Found '@babel/register' package, auto-compiling files with Babel
[0-0] 2021-02-15T17:29:10.082Z DEBUG @wdio/local-runner:utils: init multiremote session
[0-0] 2021-02-15T17:29:10.092Z DEBUG @wdio/utils:initialiseServices: initialise service "selenium-standalone" as NPM package
[0-0] 2021-02-15T17:29:10.093Z DEBUG @wdio/utils:initialiseServices: initialise service "appium" as NPM package
[0-0] RUNNING in MultiRemote - /src/test/specs/spec-android/CrossPlatform_Suite.js
[0-0] 2021-02-15T17:29:10.200Z DEBUG @wdio/local-runner:utils: init multiremote session
[0-0] 2021-02-15T17:29:10.202Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2021-02-15T17:29:10.202Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2021-02-15T17:29:10.202Z ERROR @wdio/runner: Error: Invalid or unsupported WebDriver capabilities found ("protocol", "hostname", "port", "path"). Ensure to only use valid W3C WebDriver capabilities (see https://w3c.github.io/webdriver/#capabilities).
    at Object.startWebDriverSession (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/webdriver/build/utils.js:38:19)
    at Function.newSession (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/webdriver/build/index.js:56:59)
    at Object.exports.remote (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/webdriverio/build/index.js:56:43)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at /Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/webdriverio/build/index.js:82:26
    at async Promise.all (index 0)
    at Object.exports.multiremote (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/webdriverio/build/index.js:81:5)
    at Object.initialiseInstance (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/@wdio/runner/build/utils.js:69:21)
    at Runner._startSession (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/@wdio/runner/build/index.js:198:56)
    at Runner._initSession (/Users/amrka/Desktop/work/tracker/Appium-wdio/node_modules/@wdio/runner/build/index.js:160:25)
2021-02-15T17:29:10.320Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code 1
[0-0] FAILED in MultiRemote - /src/test/specs/spec-android/CrossPlatform_Suite.js
2021-02-15T17:29:10.321Z INFO @wdio/cli:launcher: Run onComplete hook
2021-02-15T17:29:10.321Z INFO @wdio/selenium-standalone-service: shutting down all browsers
2021-02-15T17:29:10.321Z DEBUG @wdio/appium-service: Appium (pid: 2206) killed
2021-02-15T17:29:10.321Z DEBUG @wdio/cli:utils: Finished to run "onComplete" hook in 0ms
```

**Comment by christian-bromann** [comment]
Proposed a PR with a fix. Let's continue convo there.

**Comment by pablocoletto** [comment]
Issue still appears while trying to run setting up 'app' capability in an integration with Browserstack. This is the error log:

`2021-06-07T21:03:16.430Z INFO @wdio/local-runner: Start worker 0-0 with arg: src/config/wdio.conf.ts
[0-0] 2021-06-07T21:03:16.853Z INFO @wdio/local-runner: Run worker command: run
[0-0] 2021-06-07T21:03:17.041Z DEBUG @wdio/config:utils: Found 'ts-node' package, auto-compiling TypeScript files
[0-0] 2021-06-07T21:03:17.154Z DEBUG @wdio/local-runner:utils: init remote session
[0-0] 2021-06-07T21:03:17.156Z DEBUG @wdio/utils:initialiseServices: initialise service "appium" as NPM package
[0-0] 2021-06-07T21:03:17.162Z DEBUG @wdio/utils:initialiseServices: initialise service "browserstack" as NPM package
[0-0] 2021-06-07T21:03:17.168Z DEBUG @wdio/sync: Finished to run "beforeSession" hook in 0ms
[0-0] RUNNING in bs://<app ID> - /src/features/login.feature
[0-0] 2021-06-07T21:03:17.457Z DEBUG @wdio/local-runner:utils: init remote session
[0-0] 2021-06-07T21:03:17.459Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2021-06-07T21:03:17.459Z ERROR @wdio/runner: Error: Invalid or unsupported WebDriver capabilities found ("app"). Ensure to only use valid W3C WebDriver capabilities (see https://w3c.github.io/webdriver/#capabilities).`

Not sure if it may be related to the same issue mentioned above (maybe a missing capability) or a separate issue.

Thanks, Pablo

**Comment by christian-bromann** [comment]
@pablocoletto if `app` is BS specific maybe it needs to be `bs:app` then? I recommend to reach out to their support as `app` just like that violates the WebDriver protocol.

**Comment by alliv8** [comment]
Hi @christian-bromann , `app` is an [appium capability](https://appium.io/docs/en/writing-running-appium/caps/#general-capabilities) and not specific to BS.

Is handling 'appium' specific capabilities out of the box part of WebdriverIO's roadmap for W3C formatted caps?

@pablocoletto if you're still facing issues with w3c formatted caps, refer the below config object:
```
capabilities: [{
    'appium:deviceName': 'Google Pixel 3',
    'appium:osVersion': '9.0',
    'platformName': 'Android',
    'appium:app': process.env.BROWSERSTACK_APP_ID || 'bs://<APP ID>',    
    'bstack:options': {
      debug: true,
      buildName: 'Webdriverio Android',
      projectName: 'First Webdriverio Android Project',
      sessionName: 'first_test'
    }
  }]
```

There is a mix of `appium:`, `bstack:options` and root level w3c formatted property names ~

**Comment by christian-bromann** [comment]
> Is handling 'appium' specific capabilities out of the box part of WebdriverIO's roadmap for W3C formatted caps?

It is already supported, you have to use `appium:app`

**Comment by alliv8** [comment]
My question was more around if WebdriverIO would auto handle prepending of `appium:xyz` to appium specific capabilities as it is auto handled in other appium client bindings. Currently, across appium documentation, they directly mention the use of these capabilities without the `appium` keyword.

For example, the appium-python-client handles this under the hood.

The JS client [wd](https://www.npmjs.com/package/wd) listed on Appium's website, however isn't actively maintained and points towards WebdriverIO instead :).

Hence the question if WebdriverIO would extend similar functionality, or have more explicit documentation around appium capabilties as client bindings stop support for JSON Wire caps ~

**Comment by HarryFree** [comment]
I am trying to add `chromedriverExecutable: "/usr/local/bin/chromedriver"` and got an error. Is it related to this thread? Please help on this.

```
ERROR @wdio/runner: Error: Invalid or unsupported WebDriver capabilities found ("chromedriverExecutable")
```

**Comment by christian-bromann** [comment]
@HarryFree if you like to set a custom Chromedriver path, please see [the docs](https://webdriver.io/docs/wdio-chromedriver-service#chromedrivercustompath).

**Comment by HarryFree** [comment]
@christian-bromann Thank you for pointing this. But I would like to set in appium service. Do you have any reference document please?

(Launching chrome browser in mobile emulator.)

**Comment by HarryFree** [comment]
@christian-bromann I got the solution and it is working fine.

```
await browser.startActivity('com.android.chrome', 'com.google.android.apps.chrome.Main')
await browser.switchContext('WEBVIEW_chrome')
```

after start activity, i changed the context and no errors now. But I got another challenge. 

1. Open the url in the mobile chrome browser
2. We have a button `Continue to app` after launching the url. Manually When click on it, it is opening the app but it is not working through the script. I debugged it and it is actually clicking but failed to open the app.
